network/p2p/config/gossipsub.go

```diff
@@ -89,6 +94,32 @@ type ScoringParameters struct {
 	ScoringRegistryParameters ScoringRegistryParameters `validate:"required" mapstructure:"scoring-registry"`
 }
 
+// PeerGaterParameters are the parameters for the libp2p peer gater. This config provides operators the ability
+// to override topic delivery weights for the peer gater. The default weight is 1.0.
+// Parameters are "numerical values" that are used to compute or build components that compute the score of a peer in GossipSub system.
+type PeerGaterParameters struct {
+	// Enabled enables the peer gater.
+	Enabled bool `validate:"required" mapstructure:"enable"`
+	// TopicDeliveryWeightsOverride map of topic delivery weights that will override the default value for the specified channel.
+	TopicDeliveryWeightsOverride TopicDeliveryWeightsOverride `validate:"required" mapstructure:"topic-delivery-weights-override"`
+}
+
+// TopicDeliveryWeightsOverride topic delivery weights used to override the default topic delivery weight 1.0 of the peer gater.
+// Parameters are "numerical values" that are used to compute or build components that compute the score of a peer in GossipSub system.
+type TopicDeliveryWeightsOverride struct {
```

Comment:
would it be difficult to allow arbitrary strings here instead of a fixed list? It might be useful to be able to add overrides for other channels without having to rollout a new image

Reply:
https://github.com/onflow/flow-go/pull/6479/commits/02fadc667602d79b84ce47d243c4e537c3a50159